src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -585,10 +585,13 @@ def _execute(self, all_targets):
                           # Re-run tests when the code they test (and depend on) changes.
                           invalidate_dependents=True) as invalidation_check:
 
+      all_test_tgts, invalid_test_tgts = [], []
```

Comment:
NB: This change is needed because of the `partition = ...` above. It turns out that line and the associated comment are not needed / bogus / redundant to all targets being passed to coverage engine constructors regardless. A subsequent change introducing `--fast/--no-fast` partitioning in `PytestRun` style will eliminate this, but the unit test and associated changes will still stand and confirm this same problem does not creep back in... so I'd like to keep this bit of the change as-is to get caching under coverage working as a unit.